test/integration/ingress_test.go

```diff
@@ -934,3 +934,218 @@ func TestIngressRegexPrefix(t *testing.T) {
 		return false
 	}, ingressWait, waitTick)
 }
+
+func TestIngressRecoverFromInvalidPath(t *testing.T) {
+	ns, cleaner := setup(t)
+	defer func() {
+		if t.Failed() {
+			output, err := cleaner.DumpDiagnostics(ctx, t.Name())
+			t.Logf("%s failed, test ns %s, dumped diagnostics to %s", t.Name(), ns.Name, output)
+			assert.NoError(t, err)
+		}
+		assert.NoError(t, cleaner.Cleanup(ctx))
+	}()
+
+	// TODO: run this separately, make it not to affect other tests for sharing Kong.
+	t.Skipf("the case %s should be run separately", t.Name())
```

Comment:
Do I understand that correctly that during the test run we could affect other integration tests because it makes the controller stuck when applying the faulty regex path, and unblocks it after the path gets removed? 

 Should it become an E2E test to not affect other integration tests?

Reply:
> Do I understand correctly that we could affect other integration tests during the test run because it makes the controller stuck when applying the faulty regex path and unblocks it after the path gets removed?
Yes. 
> Should it become an E2E test to not affect other integration tests?
I thought about adding such an e2e test, but it is not as easy to modify a env in Kong since e2e tests uses kutomization to deploy Kong. Will do that later.

Reply:
https://github.com/Kong/kubernetes-ingress-controller/blob/c9b3bb4ed985b80096f287ca887df52cc161999b/test/e2e/kustomizations.go#L131-L154 is available. We could handle the Kong container either by making the container-specific functions (`addControllerEnv`, a new `addKongEnv`) functions that call a generic env patch helper with Deployment/Namespace/Container args or by just using a generic function directly. We only have one use of the dedicated function so far, so refactoring around a more generic signature shouldn't take much work.

Reply:
There is something different. In manifests we already defined variables `KONG_ROUTER_FLAVOR`, but kustomize does not support replacing environment: https://github.com/kubernetes-sigs/kustomize/issues/4337. Also, when we apply `add` operation, the new `KONG_ROUTER_FLAVOR` is not added at the last, so the added env could not take effect in the container.